header/p2p/server.go

```diff
@@ -52,19 +54,27 @@ func (serv *ExchangeServer) Stop(context.Context) error {
 
 // requestHandler handles inbound ExtendedHeaderRequests.
 func (serv *ExchangeServer) requestHandler(stream network.Stream) {
+	err := stream.SetReadDeadline(time.Now().Add(readDeadline))
+	if err != nil {
+		log.Warn(err)
+	}
 	// unmarshal request
 	pbreq := new(p2p_pb.ExtendedHeaderRequest)
-	_, err := serde.Read(stream, pbreq)
+	_, err = serde.Read(stream, pbreq)
 	if err != nil {
 		log.Errorw("server: reading header request from stream", "err", err)
 		stream.Reset() //nolint:errcheck
 		return
 	}
+	if err = stream.CloseRead(); err != nil {
+		log.Warn(err)
+	}
 	// retrieve and write ExtendedHeaders
-	if pbreq.Hash != nil {
-		serv.handleRequestByHash(pbreq.Hash, stream)
-	} else {
-		serv.handleRequest(pbreq.Origin, pbreq.Origin+pbreq.Amount, stream)
+	switch pbreq.Data.(type) {
```

Comment:
default case should log out error and return

Reply:
or panic 😄  as we are expecting only these two cases

Reply:
No, not panic. It's possible there will be garbage requests from the network, we should definitely not panic in that instance as that's a DoS vector (similar to the panic in the proof unmarshaller thing in fraudsync 😅 )

Reply:
ok